packages/e2e/test/wallet/SingleAddressWallet/txChainHistory.test.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable max-len */
+import { assertTxIsValid } from '../../../../wallet/test/util';
+import { buildTx } from '@cardano-sdk/wallet';
+import { deepEquals, isNotNil } from '@cardano-sdk/util';
+import { env } from '../environment';
+import { filter, firstValueFrom, map, take } from 'rxjs';
+import { getWallet } from '../../../src';
+import { logger } from '@cardano-sdk/util-dev';
+
+export type TestWallet = Awaited<ReturnType<typeof getWallet>>;
+
+describe('tx chain history', () => {
+  let wallet: TestWallet;
+
+  beforeEach(async () => {
+    wallet = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
+  });
+
+  afterEach(() => {
+    wallet.wallet.shutdown();
+  });
+
+  it('submit a transaction and find it in chain history', async () => {
+    const tAdaToSend = 10_000_000n;
+    const saWallet = wallet.wallet;
+
+    await firstValueFrom(saWallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(saWallet.addresses$);
+    const [{ address: receivingAddress }] = await firstValueFrom(saWallet.addresses$);
+
+    logger.info(
+      `Address ${sendingAddress.toString()} will send ${tAdaToSend} lovelace to address ${receivingAddress.toString()}.`
+    );
+
+    // Send 10 tADA to the same wallet.
+    const txBuilder = buildTx({ logger, observableWallet: saWallet });
+    const txOutput = txBuilder.buildOutput().address(receivingAddress).coin(tAdaToSend).toTxOut();
+    const unsignedTx = await txBuilder.addOutput(txOutput).build();
+
+    assertTxIsValid(unsignedTx);
+
+    const signedTx = await unsignedTx.sign();
+    await signedTx.submit();
+
+    logger.info(
+      `Submitted transaction id: ${signedTx.tx.id}, inputs: ${signedTx.tx.body.inputs} and outputs:${signedTx.tx.body.outputs}.`
+    );
+
+    // Wait until wallet two is aware of the funds.
+    await firstValueFrom(saWallet.balance.utxo.total$.pipe(filter(({ coins }) => coins >= tAdaToSend)));
```

Comment:
This seems to be redundant as you're awaiting for tx and not asserting the balance
```suggestion
```